src/Featurizers/Components/StandardDeviationUpdater.h

```diff
@@ -0,0 +1,156 @@
+
+    void update(InputType input);
+
+    STDResult commit(void);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    std::double_t const          _average;
+    std::uint64_t                _counter;
+    L2NormUpdater<std::double_t> _updater;
+};
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  Details::STDUpdater::STDResult
+// |
+// ----------------------------------------------------------------------
+
+template <typename T>
+Details::STDUpdater<T>::STDResult::STDResult(long double standard_deviation, std::uint64_t counter) :
+    StandardDeviation(std::move(standard_deviation)),
+    Count(std::move(counter)) {
+        // input type can only be integer or numeric
+        static_assert(Traits<T>::IsIntOrNumeric::value, "Input type to standard deviation updater has to be integer or numerical types!");
+        if (StandardDeviation < 0) {
+            throw std::invalid_argument("Standard deviation shouldn't be less than 0!");
+        }
+        if ((Count == 0) && (StandardDeviation != 0)) {
+            throw std::invalid_argument("Standard deviation is not 0 when count is 0!");
+        }
+}
+
+// ----------------------------------------------------------------------
+// |
+// |  Details::STDUpdater
+// |
+// ----------------------------------------------------------------------
+
+template <typename T>
+Details::STDUpdater<T>::STDUpdater(std::double_t average) :
+    _average(average),
+    _counter(0),
+    _updater(L2NormUpdater<std::double_t>()) {
+        // input type can only be integer or numeric
+        static_assert(Traits<T>::IsIntOrNumeric::value, "Input type to standard deviation updater has to be integer or numerical types!");
+}
+
+template <typename T>
+void Details::STDUpdater<T>::update(T input) {
+    _updater.update_l2(static_cast<std::double_t>(input) - _average);
+    // check if count will be out of bounds
+    if (std::numeric_limits<std::uint64_t>::max() == _counter) {
+        throw std::runtime_error("Overflow occured for count during calculating standard deviation! You might input too much data");
+    }
+    ++_counter;
+}
+
+template <typename T>
+typename Details::STDUpdater<T>::STDResult Details::STDUpdater<T>::commit(void) {
+    long double const l2(_updater.commit_l2().L2_norm);

```

Comment:
Throw if _counter == 0.

Reply:
I don't think this necessary since _updater.commit() will do the check to see if _counter is 0.